source/s3_meta_request.c

```diff
@@ -1764,21 +1764,33 @@ bool aws_s3_meta_request_body_has_no_more_data(const struct aws_s3_meta_request
 void aws_s3_meta_request_result_setup(
     struct aws_s3_meta_request *meta_request,
     struct aws_s3_meta_request_result *result,
-    struct aws_s3_request *request,
+    struct aws_s3_request *failed_request,
     int response_status,
     int error_code) {
 
-    if (request != NULL) {
-        if (request->send_data.response_headers != NULL) {
-            result->error_response_headers = request->send_data.response_headers;
+    if (failed_request != NULL) {
+        if (failed_request->send_data.response_headers != NULL) {
+            result->error_response_headers = failed_request->send_data.response_headers;
             aws_http_headers_acquire(result->error_response_headers);
         }
 
-        if (request->send_data.response_body.capacity > 0) {
+        if (failed_request->send_data.response_body.capacity > 0) {
             result->error_response_body = aws_mem_calloc(meta_request->allocator, 1, sizeof(struct aws_byte_buf));
 
             aws_byte_buf_init_copy(
-                result->error_response_body, meta_request->allocator, &request->send_data.response_body);
+                result->error_response_body, meta_request->allocator, &failed_request->send_data.response_body);
+        } else if (
+            failed_request->request_tag == AWS_S3_AUTO_RANGE_GET_REQUEST_TYPE_HEAD_OBJECT &&
```

Comment:
do not use `AWS_S3_AUTO_RANGE_GET_REQUEST_TYPE_HEAD_OBJECT` directly.

For different type of meta request, the `request_tag` may result in the same enum value, but have different meaning. EG: only for auto_ranged_get, 0 means HEAD_OBJECT, for auto_ranged_put, 0 means List_part. 

Use `get_request_type` vfunction on the request to get the `enum aws_s3_request_type ` that shared across different meta request type instead.

Reply:
Or double check your meta request type to make sure it's a auto_ranged_get